linkerd/app/outbound/src/http/logical.rs

```diff
@@ -147,7 +154,24 @@ impl<E> Outbound<E> {
             )
             .instrument(|l: &Logical| debug_span!("logical", dst = %l.addr()))
             .push_switch(
-                Logical::or_endpoint(tls::NoClientTls::NotProvidedByServiceDiscovery),
+                |logical: Logical| {
+                    let should_resolve = match logical.profile.as_ref() {
+                        Some(p) => {
+                            let p = p.borrow();
+                            p.endpoint.is_none() && (p.name.is_some() || !p.targets.is_empty())
+                        }
+                        None => false,
+                    };
```

Comment:
seems like this could still be abstracted over by a method on `Logical`?

Reply:
let's see where it goes -- I'm trying to decouple all of this from `Logical`.